web/src/components/Head/KotsPageTitle.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { useParams } from "react-router-dom";
+import { Helmet } from "react-helmet";
+
+import { KotsParams } from "@types";
+
+function makePageTitle({
+  appSlug,
+  pageName,
+}: {
+  appSlug?: string;
+  pageName: string;
+}): string {
+  if (appSlug) {
+    return `${pageName} | ${appSlug} | Admin Console`;
+  }
+
+  return `${pageName} | Admin Console`;
+}
+
+/*
+  example output:
+  Troubleshoot | http-echo | Admin Console
+  Gitops | Admin Console
+*/
+function KotsPageTitle({
+  pageName,
+  showAppSlug,
+}: {
+  pageName: string;
+  showAppSlug: boolean;
+}) {
+  const { slug } = useParams<KotsParams>();
+
+  if (slug && showAppSlug) {
+    return (
+      <Helmet>
+        <title>{makePageTitle({ appSlug: slug, pageName })}</title>
+      </Helmet>
+    );
+  }
+
+  return (
+    <Helmet>
+      <title>{makePageTitle({ pageName })}</title>
+    </Helmet>
+  );
```

Comment:
`react-helmet` will use the title in the farthest leaf node. So you can define a title in a parent component and in its child components and it will render the child component title when the child component is rendered.

Reply:
this is so cool